src/data-types/wrappers.h

```diff
@@ -0,0 +1,326 @@
+#pragma once
+
+#include <errno.h>
+#include <stdarg.h>
+
+#ifdef EOF // stdio.h
```

Comment:
Maybe you could just include <stdio.h> (and same for other locations)?
It would make this file simpler.
It's not an issue for me if it takes slightly more time to compile.